test/functional/feature_loadblock.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test loadblock option
+
+Test the option to start a node with the option loadpath which
+loads a serialized blockchain from a file (usually called bootstrap.dat).
+To generate that file this test uses the helper scripts in available
+in contrib/linearize.
+"""
+
+import os
+import shutil
+import urllib
+
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    initialize_datadir,
+)
+from test_framework.util import assert_equal, wait_until
+
+
+class LoadblockTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.nodes[0].generate(100)
+
+        # parsing the url of our node to get settings for config file
+        data_dir = self.nodes[0].datadir
+        node_url = urllib.parse.urlparse(self.nodes[0].url)
+        cfg_path = os.path.join(data_dir, "linearize.cfg")
+        hashlist_path = os.path.join(data_dir, "hashlist.txt")
+        bootstrap_file = os.path.join(self.options.tmpdir, "bootstrap.dat")
+        genesis_block = self.nodes[0].getblockhash(0)
+        block_dir = os.path.join(data_dir, "regtest", "blocks")
+
+        self.log.info("Create linearization config file")
+        cfg = open(cfg_path, "a", encoding="utf-8")
+        cfg.write("""datadir={datadir}
+rpcuser={user}
+rpcpassword={password}
+port={port}
+host={host}
+output_file={output_file}
+max_height=100
+netmagic=fabfb5da
+input={block_dir}
+genesis={genesis}
+hashlist={hashlist}""".format(datadir=data_dir,
+                              output_file=bootstrap_file,
+                              user=node_url.username,
+                              password=node_url.password,
+                              port=node_url.port,
+                              host=node_url.hostname,
+                              hashlist=hashlist_path,
+                              block_dir=block_dir,
+                              genesis=genesis_block))
+        cfg.close()
```

Comment:
Can use https://docs.python.org/3/reference/compound_stmts.html#with instead?

And the writing it like this, should make the test shorter in line count. https://github.com/bitcoin/bitcoin/blob/7b701fef58f627956d597817a1f9422edd890cdc/test/functional/test_framework/util.py#L306

Reply:
Yes, that's much better, implemented that.